### DISCUSSION TITLE
是否可以加入文件搜索功能

### DISCUSSION THREAD

**Discussion by DeepFal** [AlistGo/alist#553] [Ideas]
如题，如果是点进目录才加载该目录列表所以没法全局搜索的话，搜索当前目录下的文件也可以呀
文件太多了找起来很困难
谢谢

**Comment by a05332481** [comment]
把文件位置添加到数据库加上搜索就完美了

**Reply by xhofe** [reply]
v1就是这样做的 体验并不好 有文件改动的时候需要手动同步数据库 且文件多的时候同步一次数据库时间很长。

**Comment by xhofe** [comment]
当前目录可以直接使用浏览器的搜索啊。。

**Comment by wellgone** [comment]
如果是百度网盘的话，可以利用百度网盘客户端的目录缓存文件，将路径、名称、大下和类型往项目数据库里的x_search_files表导（路径需要根据自己的设置改造），手动导入后记得重启项目，否则不生效。
搜索读秀全库，完美【手动狗头】

**Comment by xuchuang199636** [comment]
关于 AList 的文件搜索功能，我目前的间接实现方案是，**RaiDrive + Everything**。相当于使用本地**文件资源管理器**作为 **AList 的第三方客户端**。

具体做法是：
1. 在 **RaiDrive** 的“**添加 → NAS → WebDAV**”界面挂载 AList 为网络磁盘。
2. 在 **Everything** 的"**工具 → 选项 → 索引 → 文件夹**"界面添加挂载的目录。
3. 等待 Everything 扫描完成，即可使用 Everything 的文件搜索功能间接实现 AList 的文件搜索功能。

相对于 AList 浏览器客户端，RaiDrive 的**优点**是：
1. 可以灵活的对文件目录进行排序。
2. 可以方便地与本机的其他优秀工具软件结合使用，比如可以使用 Typora 查看和编辑 Markdown 文件；可以使用 Everything 的检索功能。

一些**优化体验**的做法：
1. 将“**302重定向**”改为**代理**。这样能让AList更好的提供 **WebDAV** 的功能。
4. 将AList的“缓存时间”调成0（非必须，但调整之后，可以更好的使用 RaiDrive 的缓存自动刷新功能），将 RaiDrive 的“**设置 → 持续时间**”的读取和写入均调成“23:59:59”。**AList的缓存功能**仅仅是缓存目录结构，并且不能自动定时刷新。而 **RaiDrive 的缓存功能**比当前 AList 自带的缓存功能要好（个人观点），不仅能缓存目录结构，还可以缓存与 AList 服务器互传的文件，并且能够配置自动刷新。详见 [**Temporary - RaiDrive Document**](https://docs.raidrive.com/settings/temporary/) 和 [**Update storage changes asynchronously - RaiDrive Document**](https://docs.raidrive.com/advancedoptions/asynchronously/)。目前RaiDrive缓存功能的缺点是，重启电脑之后缓存会被清除。
